confidential-datahub/kms/src/plugins/aliyun/client.rs

```diff
@@ -0,0 +1,355 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::collections::BTreeMap;
+
+use async_trait::async_trait;
+use base64::engine::general_purpose::STANDARD;
+use base64::Engine;
+use chrono::Utc;
+use log::error;
+use prost::Message;
+use reqwest::Certificate;
+use reqwest::{header::HeaderMap, ClientBuilder};
+use serde_json::Value;
+use sha2::{Digest, Sha256};
+use tokio::fs;
+
+use crate::plugins::aliyun::client::dkms_api::{DecryptRequest, EncryptRequest};
+use crate::{Decrypter, Encrypter, OperationContext, ProviderSetting};
+use crate::{Error, Result};
+
+use super::annotations::{CipherContext, KmsContext};
+use super::credential::Credential;
+
+pub mod dkms_api {
+    tonic::include_proto!("dkms_api");
+}
+
+pub struct AliyunKmsClient {
+    http_client: reqwest::Client,
+    credential: Credential,
+    kms_instance_id: String,
+    endpoint: String,
+}
+
+const IN_POD_DEFAULT_KEY_PATH: &str = "/run/kms-credential/aliyun";
```

Comment:
Maaaybe there's a better place for this string (if not now , then later)? Will there be more plugins/clients in the future that reuse `/run/kms-credential`?

Reply:
Yes. Thanks for pointing out this. My idea is all KMSes should be placed in a common parent directory. To follow up the ttrpc sock used by attestation-agent https://github.com/kata-containers/kata-containers/blob/CCv0/src/agent/src/image_rpc.rs#L31, I think `/run/confidential-containers/cdh/kms-credential` might be a good place?

different kms plugins can use different sub directories.